From 9ec391e6d3f7778d2aa13e27dc1aa46ac002055a Mon Sep 17 00:00:00 2001
From: Vilas R K <vilas.r.k@intel.com>
Date: Tue, 1 Aug 2023 13:57:49 +0530
Subject: [PATCH] MIXIN changes for AIDL Conversion of Thermal HAL

Below are the details on this change.
1. caas/framework_manifest.xml & caas/manifest.xml
   --> Remove unused Thermal HIDL HAL
   Entry from framework_manifest.xml.
2. ../thermal/hal/product.mk
   Adding Thermal AIDL HAL in PRODUCT PACKAGES

Tracked-On: OAM-110601
Signed-off-by: Vilas R K <vilas.r.k@intel.com>
---
 groups/device-specific/caas/framework_manifest.xml | 13 ++++---------
 groups/device-specific/caas/manifest.xml           | 10 ----------
 groups/thermal/hal/product.mk                      |  4 +---
 groups/thermal/thermal-daemon/product.mk           |  2 +-
 4 files changed, 6 insertions(+), 23 deletions(-)

diff --git a/groups/device-specific/caas/framework_manifest.xml b/groups/device-specific/caas/framework_manifest.xml
index 4275e1e..c1fd8f1 100644
--- a/groups/device-specific/caas/framework_manifest.xml
+++ b/groups/device-specific/caas/framework_manifest.xml
@@ -47,17 +47,12 @@
         <version>1</version>
         <fqname>IHealth/default</fqname>
     </hal-->
-    <hal format="hidl">
+    <hal format="aidl">
         <name>android.hardware.thermal</name>
-        <transport>hwbinder</transport>
-        <version>1.0</version>
-        <version>2.0</version>
-        <interface>
-            <name>IThermal</name>
-            <instance>default</instance>
-        </interface>
+        <version>1</version>
+        <fqname>IThermal/default</fqname>
     </hal>
-    <hal format="aidl">
+     <hal format="aidl">
         <name>android.hardware.gatekeeper</name>
         <version>1</version>
         <interface>
diff --git a/groups/device-specific/caas/manifest.xml b/groups/device-specific/caas/manifest.xml
index fc0eec8..81d2694 100644
--- a/groups/device-specific/caas/manifest.xml
+++ b/groups/device-specific/caas/manifest.xml
@@ -9,16 +9,6 @@
         <version>1</version>
         <fqname>IHealth/default</fqname>
     </hal-->
-    <hal format="hidl">
-        <name>android.hardware.thermal</name>
-        <transport>hwbinder</transport>
-        <version>1.0</version>
-        <version>2.0</version>
-        <interface>
-            <name>IThermal</name>
-            <instance>default</instance>
-        </interface>
-    </hal>
     <hal format="hidl">
         <name>android.hardware.audio</name>
         <transport>hwbinder</transport>
diff --git a/groups/thermal/hal/product.mk b/groups/thermal/hal/product.mk
index 4e56d53..ca965f9 100644
--- a/groups/thermal/hal/product.mk
+++ b/groups/thermal/hal/product.mk
@@ -1,4 +1,2 @@
 # Thermal Hal
-PRODUCT_PACKAGES += thermal.$(TARGET_BOARD_PLATFORM) \
-                    android.hardware.thermal@2.0-service.intel \
-                    android.hardware.thermal@2.0-impl.intel
+PRODUCT_PACKAGES += android.hardware.thermal@aidl-service.intel
diff --git a/groups/thermal/thermal-daemon/product.mk b/groups/thermal/thermal-daemon/product.mk
index b77f6db..d0dddf9 100644
--- a/groups/thermal/thermal-daemon/product.mk
+++ b/groups/thermal/thermal-daemon/product.mk
@@ -5,4 +5,4 @@ PRODUCT_COPY_FILES += \
 	$(INTEL_PATH_COMMON)/thermal/thermal-daemon/thermal-cpu-cdev-order.xml:/vendor/etc/thermal-daemon/thermal-cpu-cdev-order.xml
 
 # Thermal Hal
-PRODUCT_PACKAGES += android.hardware.thermal@2.0-service.intel
+PRODUCT_PACKAGES += android.hardware.thermal@aidl-service.intel
-- 
2.17.1

